src/main/java/com/labsynch/labseer/utils/SecurityUtils.java

```diff
@@ -75,7 +76,30 @@ private Author createAuthor(Authentication authentication) {
 		
 		Author author = new Author();
 
-		if (principal instanceof LdapUserDetails) {
+		if (principal instanceof InetOrgPerson) {
+			InetOrgPerson inetOrgPerson = (InetOrgPerson) principal;
```

Comment:
Actually it looks like `InetOrgPerson` offers `getSn` and `getGivenName` methods. I think it'd be better if we tried those, then fell back to splitting Cn if we couldn't get those attributes.

Reply:
Yeah, I checked on that. 

This is our current version:
https://docs.spring.io/spring-security/site/docs/3.2.5.RELEASE/apidocs/org/springframework/security/ldap/userdetails/InetOrgPerson.html

I was just trying to get this fixed for a specific case where the CN wasn't available in the DN.  But I can try updating our spring security to see if that gets us all the way there with this fix.